protocols/imap/src/main/java/org/apache/james/imap/decode/parser/SearchCommandParser.java

```diff
@@ -74,68 +75,97 @@ protected SearchKey searchKey(ImapSession session, ImapRequestLineReader request
         } else if (next == '(') {
             return paren(session, request, charset);
         } else {
-            final int cap = consumeAndCap(request);
-            switch (cap) {
-            
-            case 'A':
-                return a(request);
-            case 'B':
-                return b(request, charset);
-            case 'C':
-                return c(session, request, isFirstToken, charset);
-            case 'D':
-                return d(request);
-            case 'E':
-                throw new DecodingException(HumanReadableText.ILLEGAL_ARGUMENTS, "Unknown search key");
-            case 'F':
-                return f(request, charset);
-            case 'G':
-                throw new DecodingException(HumanReadableText.ILLEGAL_ARGUMENTS, "Unknown search key");
-            case 'H':
+            return parseSearchKey(session, request, charset, isFirstToken);
+        }
+    }
+
+    private SearchKey parseSearchKey(ImapSession session, ImapRequestLineReader request, Charset charset, boolean isFirstToken) throws DecodingException {
+        String searchKey = request.atom().toUpperCase(Locale.US);
```

Comment:
To mitigate the issue that @mbaechler noticed, you can get only the first X characters ( where X is the size of the biggest key name) and stop at the first blank character.

Reply:
Yes, that is my MaximumSize char validator proposal.